src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int NUMBER_OF_FIGURES = 5;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureIndex = random.nextInt(NUMBER_OF_FIGURES);
+        String color = colorSupplier.getRandomColor().toLowerCase();
+        double figureProperty = random.nextDouble();
```

Comment:
The `random.nextDouble()` generates a value between 0.0 and 1.0. Ensure that this range is appropriate for the properties of the figures being created.